app/assets/javascripts/shared/dradis_datatable.js

```diff
@@ -302,6 +309,14 @@ class DradisDatatable {
     });
   }
 
+  setupLocalStorage(settings, data) {
+    if (localStorage.getItem(this.localStorageKey) === null) {
+      this.initialPageLoad = true;
+    }
```

Comment:
as mentioned in the init code, this is the reverse logic.

this method should have an if {} else {} structure, if it's the initial page load, then do the default columns thing, otherwise, load whatever was already selected.

Bonus points, those two actions (the default columns and the show/hide from storage) are actually the same action, with different inputs (the list of columns to show).